integration-tests/environments/react-native/harness/runner.js

```diff
@@ -107,124 +113,72 @@ function ensureSimulator(platform, deleteExisting = false) {
+    try {
+        const browser = await puppeteer.launch({ headless });
+        const page = await browser.newPage();
+        page.on("console", puppeteerLog.handleConsole);
+        await page.goto("http://localhost:8081/debugger-ui/");
+    } catch (err) {
+        if (err.message.startsWith("net::ERR_CONNECTION_REFUSED")) {
+            console.log("Metro was not ready: Retrying in 1s");
+            await new Promise(resolve => setTimeout(resolve, 1000));
+            return launchDebugger();
+        } else {
+            throw err;
+        }
     }
+}
 
-    const mochaConfig = {};
+async function run(headless) {
+    // Ensure the simulator is booted and ready for the app to start
+    ensureSimulator();
 
-    // Check if an argument for junit path was requested
-    if (junitFilePath) {
-        mochaConfig.reporter = "mocha-junit-reporter";
-        mochaConfig.reporterOptions = {
-            mochaFile: junitFilePath,
-        };
+    // Connect the debugger, right away
+    if (typeof headless === "boolean") {
+        await launchDebugger(headless);
     }
 
-    // Create and start a server
-    const server = new MochaRemoteServer(mochaConfig, {
-        // Accept connections only from the expected platform, to prevent cross-talk when both emulators are open
-        id: platform,
-        runOnConnection: isWatching,
-    });
-    await server.start();
-
-    // Spawn a react-native metro server
-    const metro = rn.async("start", "--reset-cache" /*, "--verbose"*/);
-    // Kill metro when the process is killed
-    process.on("exit", code => {
-        metro.kill("SIGHUP");
-    });
-    // Close the runner if metro closes unexpectedly
-    metro.on("close", code => {
-        console.error(`Metro server closed (code = ${code})`);
-        if (code !== 0) {
-            process.exit(code);
-        }
-    });
-
-    // Ensure the simulator is booted and ready for the app to start
-    ensureSimulator(platform);
-
     console.log("Starting the app");
-    if (platform === "android") {
+    if (PLATFORM === "android") {
         // Ask React Native to run the android app
         rn.sync("run-android", "--no-packager");
-    } else if (platform === "ios") {
+    } else if (PLATFORM === "ios") {
         // Ask React Native to run the ios app
         rn.sync("run-ios", "--no-packager", "--simulator", IOS_DEVICE_NAME);
     } else {
-        throw new Error(`Unexpected platform: '${platform}'`);
+        throw new Error(`Unexpected platform: '${PLATFORM}'`);
     }
 
-    // Set an interval that calls "react-native run-ios" every minute to make it refetch the bundle if it fails
-    const retryInterval = setInterval(() => {
-        if (platform === "ios") {
-            console.log("Retrying starting the iOS app");
-            // Ask React Native to re-run the ios app
-            rn.sync("run-ios", "--no-packager", "--simulator", IOS_DEVICE_NAME);
+    // Ask adb to output logs when process gets terminated or interrupted
```

Comment:
It might make more sense to stream the output of logcat continuously instead of waiting for the runner to exit.

Reply:
We might want to consider if there is an iOS equivalent to this.